pkgs/applications/emulators/vita3k/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchFromGitHub, cmake, ninja, clang, openssl, pkg-config, boost, python3, SDL2 }:
+
+stdenv.mkDerivation rec {
+  pname = "vita3k";
+  version = "unstable-2022-10-09";
+
+  src = fetchFromGitHub {
+    owner = "Vita3K";
+    repo = "Vita3K";
+    rev = "45bdd66556b6c6c49411c58c0aeb8f12973cf40c";
+    sha256 = "sha256-edrLeJkacKFMV4f2ytJlHp68FlygDWgxoZ7xJliqS6g=";
+    fetchSubmodules = true;
+    deepClone = true;
+  };
+
+  nativeBuildInputs = [ cmake ninja clang pkg-config ];
+
+  buildInputs = [ openssl boost python3 SDL2 ];
+
+  buildFlags = [ "--preset linux-ninja-clang" "-DUSE_VITA3K_UPDATE=OFF" "--config Release" ];
```

Comment:
You prolly want these to be `cmakeFlags` instead, and 2/3 of them might not be needed:

> `--config Release`

This is automatically passed to the CMake invocation, the config value can be controlled with the `cmakeBuildType` variable. It defaults to `Release`, so you should be able to just drop this setting.

https://github.com/NixOS/nixpkgs/blob/8ff7b290e6dd47d7ed24c6d156ba60fc3c83f100/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L111

> `--preset linux-ninja-clang`

Having `ninja` in `nativeBuildInputs` should be enough to opt into using it with CMake, all the setup hooks should take care of enabling the correct generator and calling the appropriate build command.
https://github.com/NixOS/nixpkgs/blob/8ff7b290e6dd47d7ed24c6d156ba60fc3c83f100/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L120-L122

For the `clang` part of this & in `nativeBuildInputs`, you'll likely want to use `clangStdenv` instead. That's the correct way of using a toolchain that differs from the default.

(Excuse the screenshot, GitHub has deemed [pkgs/top-level/all-packages.nix](https://github.com/NixOS/nixpkgs/blob/b28d48748830f1c3402b44d35bc3552b202eb784/pkgs/top-level/all-packages.nix) too large for any nice views & links)
![Bildschirmfoto von 2022-10-11 21-55-48](https://user-images.githubusercontent.com/23431373/195186539-f8a76c2a-c461-41e0-a5b9-a244575f5bf4.png)